packages/rum-core/src/common/apm-server.js

```diff
@@ -158,9 +164,26 @@ class ApmServer {
     if (environment) {
       configEndpoint += `&service.environment=${environment}`
     }
+
+    let localConfig = this._configService.getLocalConfig()
+    if (localConfig) {
+      configEndpoint += `&ifnonematch=${localConfig.etag}`
+    }
+
     return this._makeHttpRequest('GET', configEndpoint, { timeout: 5000 })
-      .then(response => {
-        return JSON.parse(response)
+      .then(xhr => {
+        const { status, responseText } = xhr
+        if (status === 304) {
+          return localConfig
+        } else {
+          let remoteConfig = JSON.parse(responseText)
+          const etag = xhr.getResponseHeader('etag')
+          if (etag) {
+            remoteConfig.etag = etag.replace(/["]/g, '')
```

Comment:
was there any discussion on storing the etag without double quotes in RUM? I see two contradicting comments here https://github.com/elastic/apm-server/pull/2747#discussion_r333453028

Reply:
i honestly would have preferred to keep the double quotes everywhere as-is to not need extra logic on both sides, but from previous discussion with @jahtalab , apm-server will expect the `ifnonematch` query parameter without double quotes (only when passed as a query param).

the Etag header returned by apm-server in the response will include double quotes.

Reply:
The main reason was that double quotes need to be encoded in the URL (and decoded on the server).